src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -355,6 +356,16 @@ DatabaseShardImp::importShard(
         return false;
     }
 
+    auto expectedHash =
+        app_.getLedgerMaster().walkHashBySeq(lastLedgerSeq(shardIndex));
+
+    if (!expectedHash)
+    {
+        JLOG(j_.error()) << "shard " << shardIndex
```

Comment:
Import shouldn't have been called if the expectedHash is not available. Regardless, since import can't proceed in this case I think the higher level is appropriate.